### DISCUSSION TITLE
How about add a compile option to disable encryption?

### DISCUSSION THREAD

**Discussion by Ncerzzk** [svpcom/wfb-ng#243] [Ideas]
Hi, I'm working on porting wifibroadcast TX to esp32, and hope to work with general Wifibroadcast GS.

as you know, esp32 is not a strong CPU(240 Mhz,dual core), and I hope to disable the encryption to save some CPU time.

I have not made a experiment yet, is it worth to do so?

Actually, I'm not familiar with encryption and decryption, and don't know the resource cost by them. 

If it's possiable, may be I could  help to do this changes?

**Comment by svpcom** [comment]
Disabling encryption is a bad idea. It is also used for stream consistency checking. The most resource greedy part is a FEC. If you want to send only telemetry then CPU usage will be acceptable for esp32.

**Reply by Ncerzzk** [reply]
> Disabling encryption is a bad idea. It is also used for stream consistency checking. The most resource greedy part is a FEC. If you want to send only telemetry then CPU usage will be acceptable for esp32.

Thank you. I have finished the experiment to encrypt 1000 bytes using `crypto_aead_chacha20poly1305_encrypt` in esp32. And it cost about 385us, I think the cost is affordable for esp32.

test code:
```
uint8_t data[1000]={1,2,4,5,15,5,5,5,5,5,50};
uint8_t outdata[1444]={0};
extern "C" void app_main()
{


    uint64_t outsize;
    uint64_t cnt=0;
    uint8_t key[32]={1,2,3,4,56,6};
    srand(esp_timer_get_time());

    printf("Initializing...\n");
    
    while (true)
    {
        esp_task_wdt_reset();
        uint64_t t1=micros();
        crypto_aead_chacha20poly1305_encrypt(outdata,&outsize,data,sizeof(data),NULL,0,NULL,(uint8_t *)&cnt,key);
        uint64_t t2=micros();
        cnt++;
        printf("encrypt time=%lld outsize:%lld \n",t2-t1,outsize); 
        vTaskDelay(10);
        
    }
}
```

Besides, I want to use esp32 to transmit video stream, not only telemetry data, just like 
this project:
https://github.com/jeanlemotan/esp32-cam-fpv
